src/SignalR/common/Shared/ClientResultsManager.cs

```diff
@@ -23,7 +23,7 @@ public Task<T> AddInvocation<T>(string connectionId, string invocationId, Cancel
             var tcs = (TaskCompletionSourceWithCancellation<T>)state;
             if (completionMessage.HasResult)
```

Comment:
better fix
```suggestion
            if (completionMessage.Result != null)
```

Reply:
No, `Result` is allowed to be null, the issue is using generics and nullability doesn't play nicely together.

Reply:
@BrennanConroy is the current fix OK then?

Reply:
Yes, unless you know a better way to annotate generics to allow nullability.